service/storage_proxy.cc

```diff
@@ -4225,7 +4213,12 @@ void storage_proxy::send_to_live_endpoints(storage_proxy::response_id_type respo
         local.emplace_back("", handler.get_targets());
     }
 
-    auto all = boost::range::join(local, dc_groups);
+    using dc_and_address = std::pair<const sstring, inet_address_vector_replica_set>;
+    auto all = utils::small_vector<dc_and_address*, 10>();
+    all.reserve(local.size() + dc_groups.size());
+    std::ranges::copy(local | std::views::transform([] (auto& x) { return &x; }), std::back_inserter(all));
+    std::ranges::copy(dc_groups | std::views::transform([] (auto& x) { return &x; }), std::back_inserter(all));
+
```

Comment:
we'll be able to avoid the copy once we have std::ranges::views::concat in c++26.

Reply:
We shouldn't use unordered_map for such small containers, instead put everything into utils::small_vector and apply in-place sorting and partitioning algorithms.

Reply:
And views::concat (like boost::join) is a bad idea, it has to type-erase the underlying iterator type. So it's better to refactor the whole thing.